lib/valkyrie/specs/shared_specs/locking_persister.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+RSpec.shared_examples 'a Valkyrie locking persister' do
+  before do
+    class MyLockingResource < Valkyrie::Resource
+      enable_optimistic_locking
+      attribute :id, Valkyrie::Types::ID.optional
+      attribute :title
+    end
+  end
+
+  after do
+    ActiveSupport::Dependencies.remove_constant("MyLockingResource")
+  end
+
+  describe "#save" do
+    context "when creating a resource" do
+      it "returns the value of the system-generated optimistic locking attribute on the resource" do
+        resource = MyLockingResource.new(title: ["My Locked Resource"])
+        saved_resource = persister.save(resource: resource)
+        expect(saved_resource.send(Valkyrie::Persistence::Attributes::OPTIMISTIC_LOCK)).not_to be_empty
+      end
+    end
+
+    context "when updating a resource with a correct lock token" do
+      it "successfully saves the resource and returns the updated value of the optimistic locking attribute" do
+        resource = MyLockingResource.new(title: ["My Locked Resource"])
+        initial_resource = persister.save(resource: resource)
+        updated_resource = persister.save(resource: initial_resource)
+        expect(initial_resource.send(Valkyrie::Persistence::Attributes::OPTIMISTIC_LOCK))
+          .not_to eq updated_resource.send(Valkyrie::Persistence::Attributes::OPTIMISTIC_LOCK)
+      end
+    end
+
+    context "when updating a resource with an incorrect lock token" do
+      it "raises a Valkyrie::Persistence::StaleObjectError" do
+        resource = MyLockingResource.new(title: ["My Locked Resource"])
+        initial_resource = persister.save(resource: resource)
+        initial_resource.send("#{Valkyrie::Persistence::Attributes::OPTIMISTIC_LOCK}=", [123])
+        expect { persister.save(resource: initial_resource) }.to raise_error(Valkyrie::Persistence::StaleObjectError, resource.id.to_s)
+      end
+    end
+
+    # The only error we catch is the 409 conflict
+    context "when updating a resource with an invalid token" do
+      it "raises an Rsolr 500 Error" do
```

Comment:
This seems like a spec which should be part of the solr adapter maybe? Or maybe even deleted - seems like just generally invalid behavior.

Reply:
We added this to cover the other path of the conditional. we will move it to the solr adapter, though, good catch!